### PR TITLE
Added time helper tests.

### PR DESCRIPTION
**Pull request by nventuro**
Part of #1078.

### CODE REVIEW

File: test/helpers/test/time.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  beforeEach(async function () {
+    this.start = await time.latest();
+  });
+
+  describe('increase', function () {
+    it('increases time by a duration', async function () {
+      await time.increase(time.duration.hours(1));
+
+      const end = this.start + time.duration.hours(1);
+      (await time.latest()).should.be.within(end - 20, end + 20); // +-20 sec tolerance
```
Comment (frangio) [inline test/helpers/test/time.test.js]: We can use [`closeTo`](https://www.chaijs.com/api/bdd/#method_closeto). Can we define the tolerance in a file-wide constant?

Also, I'm curious, experimentally, what did you find the actual time increase to be?

File: test/helpers/test/time.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    });
+
+    it('throws with negative durations', async function () {
+      await shouldFail(time.increase(-1));
+    });
+  });
+
+  describe('increaseTo', function () {
+    it('increases time to a time in the future', async function () {
+      await time.increaseTo(this.start + time.duration.hours(1));
```
Comment (frangio) [inline test/helpers/test/time.test.js]: Why not use the `end` variable defined below?

Comment (nventuro) [reply test/helpers/test/time.test.js]: Because it seems I'm not very smart :sweat_smile:

File: test/helpers/test/time.test.js (hunk 1 above)
Comment (nventuro) [reply test/helpers/test/time.test.js]: Didn't know about `closeTo`, thanks!

The time tolerance comes from the last block (and therefore the timestamp) being mined in the previous test, and the time between tests is variable (in particular, the time between `ganache` being set-up and the first test running may be very large). 

I ended up making this more explicit by adding an `advanceBlock` call before calling `latest()`, but we should make sure the documentation explains this behavior properly.

**Pull request merged**